server/kotc_actix/src/kots_ws_session.rs

```diff
@@ -0,0 +1,111 @@
+use std::time::{Duration, Instant};
+use actix::{Actor, ActorContext, ActorFuture, Addr, AsyncContext, ContextFutureSpawner, fut, Handler, Running, StreamHandler, WrapFuture};
+use actix_web_actors::ws;
+use actix_web_actors::ws::Message::Text;
+use rand::Rng;
+
+use crate::kotc_messages::{ClientMessage, Connect, Disconnect, KotcMessage};
+use crate::kotc_ws_server::KotcWsServer;
+
+const HEARTBEAT_INTERVAL: Duration = Duration::from_secs(5);
+const CLIENT_TIMEOUT: Duration = Duration::from_secs(10);
+
+pub struct KotcWsSession {
+    id: usize,
+    lobby_id: usize,
+    server_addr: Addr<KotcWsServer>,
+    hb: Instant,
+}
+
+impl KotcWsSession {
+    pub fn new(lobby_id: usize, server_addr: Addr<KotcWsServer>) -> Self {
+        let mut rng = rand::thread_rng();
+        KotcWsSession {
+            id: rng.gen(),
+            lobby_id,
+            server_addr,
+            hb: Instant::now(),
+        }
+    }
+
+    fn hb(&self, ctx: &mut ws::WebsocketContext<Self>) {
+        ctx.run_interval(HEARTBEAT_INTERVAL, |session, ctx| {
+            if Instant::now().duration_since(session.hb) > CLIENT_TIMEOUT {
+                println!("Disconnecting, failed heartbeat!");
+                session.server_addr.do_send(Disconnect { id: session.id, lobby_id: session.lobby_id});
+                ctx.stop();
+                return;
+            }
+
+            ctx.ping(b"Ping");
+        });
+    }
+}
+
+impl StreamHandler<Result<ws::Message, ws::ProtocolError>> for KotcWsSession {
+    fn handle(&mut self, msg: Result<ws::Message, ws::ProtocolError>, ctx: &mut Self::Context) {
+        match msg {
+            Ok(ws::Message::Ping(msg)) => {
```

Comment:
Just a thing to think about (not sure if it is better, but i like it more), how about:
Ok(message) => match message {
  ws::Message::Ping(msg) => {...},
  ws::Message::Pong(_) => {...},
  ...
}